opal/views.py

```diff
@@ -50,23 +50,46 @@ def get_context_data(self, **kwargs):
         context = super(
             PatientListTemplateView, self
         ).get_context_data(**kwargs)
-        list_slug = None
-        if self.patient_list:
-            list_slug = self.patient_list.get_slug()
-        context['list_slug'] = list_slug
         context['patient_list'] = self.patient_list
-        context['lists'] = list(PatientList.for_user(self.request.user))
-        context['num_lists'] = len(context['lists'])
+        context['switcher_items'] = self.get_switcher_items()
+        context['list_group'] = self.get_list_group_context()
+        context['columns'] = self.get_column_context(**kwargs)
+        return context
 
-        context['list_group'] = None
+    # builds a list of Switcher items filtered for user, tab groups and current item viewed
+    def get_switcher_items(self):
```

Comment:
If I'm reading the intention correctly, I might spell this something like:
```python
lists = [p for p in PatientList.for_user(self.request.user) if p != self.patient_list]
return {
    TabbedPatientListGroup.for_list(l) if TabbedPatientListGroup.for_list(l) else l for l in lists
}
```
(I've not actually tested that this works BTW)